### PR TITLE
AC-424:Do not require login location if no locations are configured.

### PR DESCRIPTION
**Pull request by anonymous-ME**
**JIRA TICKET NAME:**

[AC-424:Do not require login location if no locations are configured.](https://issues.openmrs.org/browse/AC-424)

**Summary:**
Do not require login location if no locations are configured.

### CODE REVIEW

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java [hunk 1]
```diff
@@ -353,7 +360,7 @@ private Snackbar createSnackbar(String message) {
     @Override
     public void setLocationErrorOccurred(boolean errorOccurred) {
         this.loginValidatorWatcher.setLocationErrorOccurred(errorOccurred);
-        mLoginButton.setEnabled(!errorOccurred);
```
Comment (dkayiwa) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java]: Why don't you want to disable the login button when errors occur?

File: openmrs-client/src/main/java/org/openmrs/mobile/listeners/watcher/LoginValidatorWatcher.java [hunk 2]
```diff
@@ -116,17 +116,20 @@ public void onNothingSelected(AdapterView<?> adapterView) {
 
     private boolean isAllDataValid() {
 
-        boolean result = validateNotEmpty(mUsername) && validateNotEmpty(mPassword) && validateNotEmpty(mLocation) && !urlChanged;
```
Comment (dkayiwa) [inline openmrs-client/src/main/java/org/openmrs/mobile/listeners/watcher/LoginValidatorWatcher.java]: I would think that the validateNotEmpty() check stays, but in its body, we simply return valid of there are no login locations.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java [hunk 3]
```diff
@@ -308,6 +308,13 @@ public void initLoginForm(List<Location> locationsList, String serverURL) {
         mLoginButton.setEnabled(false);
         mSpinner.setVisibility(View.GONE);
         mLoginFormView.setVisibility(View.VISIBLE);
+        if (locationsList.isEmpty()) {
+            mDropdownLocation.setVisibility(View.GONE);
+            mLoginButton.setEnabled(true);
+        }else {
+            mDropdownLocation.setVisibility(View.VISIBLE);
```
Comment (dkayiwa) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java]: Do we need the else? I thought it was already disabled by default.

**Comment by dkayiwa** [comment]
The screenshot should be on the JIRA ticket instead of pull request.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java (hunk 3 above)
Comment (anonymous-ME) [reply openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java]: @dkayiwa when the url / API is updated , the visiblity of the mDropdownLocation should change accordingly .

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java (hunk 1 above)
Comment (anonymous-ME) [reply openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java]: Will make necessary changes .

File: openmrs-client/src/main/java/org/openmrs/mobile/listeners/watcher/LoginValidatorWatcher.java (hunk 2 above)
Comment (anonymous-ME) [reply openmrs-client/src/main/java/org/openmrs/mobile/listeners/watcher/LoginValidatorWatcher.java]: Will make necessary changes .

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java [hunk 4]
```diff
@@ -353,7 +360,7 @@ private Snackbar createSnackbar(String message) {
     @Override
     public void setLocationErrorOccurred(boolean errorOccurred) {
         this.loginValidatorWatcher.setLocationErrorOccurred(errorOccurred);
-        mLoginButton.setEnabled(!errorOccurred);
+        mDropdownLocation.setVisibility(View.GONE);
```
Comment (dkayiwa) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java]: Isn't the above supposed to depend on the value of errorOccurred?

**Comment by dkayiwa** [comment]
@anonymous-ME can you squash these commits into one as advised at? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by anonymous-ME** [comment]
@dkayiwa squashed all commits into one.

**Comment by anonymous-ME** [comment]
Found a bug in this PR in validating the location dropdown. 

Please refer to this PR for this issue. 
https://github.com/openmrs/openmrs-contrib-android-client/pull/395

**Pull request merged**